consensus/hotstuff/committees/consensus_committee.go

```diff
@@ -15,38 +13,43 @@ import (
 	"github.com/onflow/flow-go/module/irrecoverable"
 	"github.com/onflow/flow-go/state/protocol"
 	"github.com/onflow/flow-go/state/protocol/events"
-	"github.com/onflow/flow-go/state/protocol/prg"
 )
 
-// staticEpochInfo contains leader selection and the initial committee for one epoch.
-// This data structure must not be mutated after construction.
-type staticEpochInfo struct {
-	firstView            uint64                  // first view of the epoch (inclusive)
-	finalView            uint64                  // final view of the epoch (inclusive)
-	randomSource         []byte                  // random source of epoch
-	leaders              *leader.LeaderSelection // pre-computed leader selection for the epoch
-	initialCommittee     flow.IdentitySkeletonList
-	initialCommitteeMap  map[flow.Identifier]*flow.IdentitySkeleton
-	weightThresholdForQC uint64 // computed based on initial committee weights
-	weightThresholdForTO uint64 // computed based on initial committee weights
-	dkg                  hotstuff.DKG
+// epochInfo caches data about one epoch that is pertinent to the consensus committee.
+type epochInfo struct {
+	*leader.LeaderSelection // pre-computed leader selection for the epoch
+	initialCommittee        flow.IdentitySkeletonList
+	initialCommitteeMap     map[flow.Identifier]*flow.IdentitySkeleton
+	weightThresholdForQC    uint64 // computed based on initial committee weights
+	weightThresholdForTO    uint64 // computed based on initial committee weights
+	dkg                     hotstuff.DKG
 }
 
-// newStaticEpochInfo returns the static epoch information from the epoch.
-// This can be cached and used for all by-view queries for this epoch.
-func newStaticEpochInfo(epoch protocol.Epoch) (*staticEpochInfo, error) {
-	firstView, err := epoch.FirstView()
+// recomputeLeaderSelectionForExtendedViewRange re-computes the LeaderSelection field
+// for the input epoch's entire view range, including extensions.
+// This should be called each time an extension is added to an epoch.
+// No errors are expected during normal operation.
```

Comment:
```suggestion
// Caution, not concurrency safe.
// No errors are expected during normal operation.
```

Reply:
I think we should be more strict in such case, specifically mention that we need to acquire lock before calling this function.

Reply:
Wondering if I am missing something. My thoughts on the matter is as follows:

`epochInfo` is only used internally in the consensus committee. Within a single component, there are different ways to guarantee concurrency safety, beyond using a lock: atomics write-read establish a ["happen-before relationship"](https://go.dev/ref/mem) (this is a well-established technique called [piggybacking](https://hub.jmonkeyengine.org/t/question-about-using-volatile-keyword-in-sio2-job/46422/2) in Java, which completely avoids lock contention), alternatively interactions with `epochInfo` being limited to a _single_ go-routine is another way of guaranteeing concurrency safety. 

Hence, I would be inclined to avoid the limitations to mutex.